packages/viz/src/lib/popups/Popup.ts

```diff
@@ -0,0 +1,282 @@
+import { Deck } from '@deck.gl/core';
+import { CartoPopupError, popupErrorTypes } from '../errors/popup-error';
+
+/**
+ * Default options for the Popup
+ */
+const defaultOptions = {
+  closeButton: true,
+  containerClassName: 'carto-popup',
+  contentClassName: 'carto-popup-content',
+  closeButtonClassName: 'carto-popup-close'
+};
+
+/**
+ * @class
+ * This class wraps the popup based on the
+ * implementation.
+ */
+export class Popup {
+  private _options: PopupOptions;
+  private _coordinates: number[] | undefined;
+  private _deckInstance: Deck | undefined;
+  private _container: HTMLElement;
+  private _parentElement: HTMLElement | undefined;
+  private _isOpened: boolean;
+
+  constructor(options: any = defaultOptions) {
```

Comment:
Let's create an interface for this:
```typescript
export interface DefaultPopupOptions {
  closeButton: boolean;
  containerClassName: string;
  contentClassName: string;
  closeButtonClassName: string;
}
```

And change the constructor to:
```typescript
constructor(options: Partial<DefaultPopupOptions> = {}) {
```

You can use Partial<PopupOptions>=

Reply:
Mmmm I see you already have the interface! Use just the partial and remove the any

Reply:
done! thanks for the tip! :rocket: